transformer/transformers/ampruntimejs_test.go

```diff
@@ -0,0 +1,70 @@
+package transformers_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/ampproject/amppackager/transformer/internal/amphtml"
+	tt "github.com/ampproject/amppackager/transformer/internal/testing"
+	"github.com/ampproject/amppackager/transformer/transformers"
+	"golang.org/x/net/html"
+)
+
+func TestStripJS(t *testing.T) {
+	tcs := []tt.TestCase{
+		{
+			Desc:     "no script node",
+			Input:    "<head></head><body></body>",
+			Expected: "<head></head><body></body>",
+		},
+		{
+			Desc:     "no prefix",
+			Input:    "<script src=main.js/>",
+			Expected: "<script src=main.js/>",
+		},
+		{
+			Desc:     "no suffix",
+			Input:    "<script src=\"https://cdn.ampproject.org\"/>",
+			Expected: "<script src=\"https://cdn.ampproject.org\"/>",
+		},
+		{
+			Desc:     "transformation",
+			Input:    "<script async src=\"https://cdn.ampproject.org/v0.js\"></script>",
+			Expected: "<script async src=\"https://cdn.ampproject.org/v0.sxg.js\"></script>",
+		},

```

Comment:
Add a test for an extension script being transformed.

Reply:
@twifkak differential loading has compelling reasons for it's existence, but I still don't see an underlying explanation of it's importance here (not saying there isn't one, just would like it described in a design document). AMP has supported `type=module/nomodule` for improved browser/domain performance. The interface here is the extension `sxg.js`. Is that the preference AMP should follow based on either extension tags or attributes within the tag? Given the contention within the issue (https://github.com/whatwg/html/issues/4432) you raised, I'd think we should be careful here and get feedback before implementing. @jridgewell as an fyi

Reply:
Using a query param was my recommended approach. A few spots in the runtime suppose that the filename will be `-VERSION.js` or `-VERSION.mjs`, so adding a `.sxg` before the extension was going to complicate matters.